taskcluster/translations_taskgraph/actions/train.py

```diff
@@ -58,6 +57,40 @@ def validate_pretrained_models(params):
             )
 
 
+def get_descendants(task_id, queue, res=None):
```

Comment:
Is it possible that this may be called with the same `task_id` more than once? I don't think so....but if it is, you should memoize this function to avoid repeated work.

Reply:
This check prevents it:

```python
       if dep_id in res:
            continue

        res[dep_id] = dep_name
        # recurse down to this child’s dependents
        get_descendants(dep_id, queue, res)
```

Reply:
Ah, indeed!